### PR TITLE
892301 - Update the Gantt UG for Custom view model Import and Export.

### PR DESCRIPTION
**Pull request by sanjayvikramsf4373**
### Description

Update the Gantt UG for Custom view model Import and Export.

### CODE REVIEW

File: wpf/Gantt/import-and-export-support.md [hunk 1]
```diff
@@ -62,7 +62,9 @@ bool</td></tr>
 </table>
 
 
-## Import/Export Task Details from/to XML
+## Import/Export Tasks from/to XML
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:65]: ## Importing and Exporting tasks from/to XML

File: wpf/Gantt/import-and-export-support.md [hunk 2]
```diff
@@ -62,7 +62,9 @@ bool</td></tr>
...
+
+### Using Default view model(Task Details)
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:67]: Using default model (TaskDetails)

File: wpf/Gantt/import-and-export-support.md [hunk 3]
```diff
@@ -266,6 +243,380 @@ private void OnOpenButtonClick(object sender, System.Windows.RoutedEventArgs e)
 {% endhighlight %}
 {% endtabs %}
 
+### Using Custom view model
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:246]: Using custom model

File: wpf/Gantt/import-and-export-support.md [hunk 4]
```diff
@@ -266,6 +243,380 @@ private void OnOpenButtonClick(object sender, System.Windows.RoutedEventArgs e)
...
+
+The following code illustrates how to Import and Export Custom ViewModel collections from or to XML.
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:248]: The following code illustrates how to Import and Export custom model from or to XML.

File: wpf/Gantt/import-and-export-support.md [hunk 5]
```diff
@@ -266,6 +243,380 @@ private void OnOpenButtonClick(object sender, System.Windows.RoutedEventArgs e)
...
+                 Content="Export To XML" />
+         <Button Height="25"
+                 HorizontalAlignment="Center"
+                 VerticalAlignment="Center"
+                 Margin="0,10,0,0"
+                 Width="200"
+                 Click="OnOpenButtonClick"
+                 Content="Import From XML" />
+</StackPanel>
+<syncfusion:GanttControl x:Name="ganttControl"
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:276]: Improper code indentation

File: wpf/Gantt/import-and-export-support.md [hunk 6]
```diff
@@ -266,6 +243,380 @@ private void OnOpenButtonClick(object sender, System.Windows.RoutedEventArgs e)
...
+                             ChildMapping="ChildCollection"
+                             FinishDateMapping="EndDate"
+                             DurationMapping="Duration"
+                             ProgressMapping="Progress"
+                             PredecessorMapping="Predecessor"
+                             ResourceInfoMapping="Resource">
+    </syncfusion:TaskAttributeMapping>
+</syncfusion:GanttControl.TaskAttributeMapping>
+    <syncfusion:GanttControl.DataContext>
+        <local:GanttViewModel/>
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:292]: Here, defined class name is GanttViewModel, but added view model class named as ViewModel. resolve this

File: wpf/Gantt/import-and-export-support.md [hunk 7]
```diff
@@ -266,6 +243,380 @@ private void OnOpenButtonClick(object sender, System.Windows.RoutedEventArgs e)
...
+      public double Progress
+      {
+          get
+          {
+              return this.progress;
+          }
+          set
+          {
+              this.progress = value;
+              OnPropertyChanged("Progress");
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:472]: Use prefix `this` in all places.

File: wpf/Gantt/import-and-export-support.md [hunk 8]
```diff
@@ -266,6 +243,380 @@ private void OnOpenButtonClick(object sender, System.Windows.RoutedEventArgs e)
...
+    <syncfusion:GanttControl.DataContext>
+        <local:GanttViewModel/>
+    </syncfusion:GanttControl.DataContext>
+</syncfusion:GanttControl>
+</Grid>
+
+{% endhighlight  %}
+
+{% highlight c# %}
+
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:300]: Remove unwanted extra lines added.

File: wpf/Gantt/import-and-export-support.md [hunk 9]
```diff
@@ -266,6 +243,380 @@ private void OnOpenButtonClick(object sender, System.Windows.RoutedEventArgs e)
...
+
+//Initializing Gantt
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:302]: Remove this comment

File: wpf/Gantt/import-and-export-support.md [hunk 10]
```diff
@@ -266,6 +243,380 @@ private void OnOpenButtonClick(object sender, System.Windows.RoutedEventArgs e)
...
+
+     public ViewModel()
+     {
+         this.TaskCollection = this.GetDataSource();
+     }
+
+     private ObservableCollection<Task> GetDataSource()
+     {
+
+         var data = new ObservableCollection<Task>();
```
Comment (Karthickmani97) [inline wpf/Gantt/import-and-export-support.md:557]: data - rename to `taskInfo`

**Review by Karthickmani97 — COMMENTED** [review]
@sanjayvikramsf4373 , Address my review concerns.

File: wpf/Gantt/import-and-export-support.md (hunk 1 above)
Comment (sanjayvikramsf4373) [reply wpf/Gantt/import-and-export-support.md]: Changed the Title.

File: wpf/Gantt/import-and-export-support.md (hunk 2 above)
Comment (sanjayvikramsf4373) [reply wpf/Gantt/import-and-export-support.md]: Changed the sub title.

File: wpf/Gantt/import-and-export-support.md (hunk 3 above)
Comment (sanjayvikramsf4373) [reply wpf/Gantt/import-and-export-support.md]: Changed the sub title.

File: wpf/Gantt/import-and-export-support.md (hunk 4 above)
Comment (sanjayvikramsf4373) [reply wpf/Gantt/import-and-export-support.md]: Changed the discription.

File: wpf/Gantt/import-and-export-support.md (hunk 5 above)
Comment (sanjayvikramsf4373) [reply wpf/Gantt/import-and-export-support.md:276]: revert the changes.

File: wpf/Gantt/import-and-export-support.md (hunk 6 above)
Comment (sanjayvikramsf4373) [reply wpf/Gantt/import-and-export-support.md]: Changed GanttViewModel to ViewModel.

File: wpf/Gantt/import-and-export-support.md (hunk 8 above)
Comment (sanjayvikramsf4373) [reply wpf/Gantt/import-and-export-support.md:300]: Removed unwanted lines added.

File: wpf/Gantt/import-and-export-support.md (hunk 9 above)
Comment (sanjayvikramsf4373) [reply wpf/Gantt/import-and-export-support.md]: Removed Comment.

**Review by Jeyasri-Murugan — APPROVED** [review]

**Pull request merged**